File: pegasusX/apps/warehouse-app-android/app/src/main/java/com/pegasusx/warehouse/ui/portal/WarehousePortalLinks.kt
package com.pegasusx.warehouse.ui.portal

import android.net.Uri
import com.pegasusx.warehouse.BuildConfig

object WarehousePortalLinks {
    fun url(feature: WarehousePortalFeature): String {
        val base = BuildConfig.PORTAL_BASE_URL.trimEnd('/')
        val path = feature.portalPath.trimStart('/')
        return if (path.isEmpty()) base else "$base/$path"
    }

    fun openUri(feature: WarehousePortalFeature): Uri = Uri.parse(url(feature))
}

File: pegasusX/apps/warehouse-app-ios/WarehouseApp/Utilities/WarehousePortalLinks.swift
import Foundation

enum WarehousePortalLinks {
    /// Dev default: warehouse-portal on port 3002. Override with `PEGASUS_PORTAL_HOST` (host:port, no scheme).
    static var baseURL: URL {
        let raw = ProcessInfo.processInfo.environment["PEGASUS_PORTAL_HOST"] ?? "localhost:3002"
        if raw.hasPrefix("http://") || raw.hasPrefix("https://") {
            return URL(string: raw.trimmingCharacters(in: CharacterSet(charactersIn: "/")))!
        }
        return URL(string: "http://\(raw)")!
    }

    static func url(for feature: WarehousePortalFeature) -> URL {
        var components = URLComponents(url: baseURL, resolvingAgainstBaseURL: false)!
        let trimmed = feature.portalPath.trimmingCharacters(in: CharacterSet(charactersIn: "/"))
        components.path = trimmed.isEmpty ? "/" : "/\(trimmed)"
        return components.url ?? baseURL
    }
}
